### PR TITLE
Add fcontext resource to manage file contexts.

### PR DESCRIPTION
**Pull request by joerg**
Sorry, I couldn't get find to work. For me, find always returned no results if the regex did include any brackets.
I used a recursive call to restorecon for now, but this can easily be changed since it is in its own helper definition.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,38 @@ allow openvpn_t openvpn_etc_t:file { write unlink };
   action :deploy
 end
 ```
+fcontext
```
Comment (BackSlasher) [inline README.md]: `### fcontext`

File: README.md [hunk 2]
```diff
@@ -141,6 +173,8 @@ The generic method seems fine to me:
 License and Authors
 -------------------
 Licensed [GPL v2](http://choosealicense.com/licenses/gpl-2.0/)  
-Author: Nitzan Raz ([backslasher](http://backslasher.net))  
+Author: Nitzan Raz ([backslasher](http://backslasher.net))
+Contributors:
```
Comment (BackSlasher) [inline README.md]: Missing newline or something, check preview (or "rich diff")

File: providers/fcontext.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  end
+end
+
+# Delete if exists
+action :delete do
+  if use_selinux then
+    escaped_file_spec = Regexp.escape(new_resource.file_spec)
+    execute "selinux-fcontext-#{new_resource.secontext}-delete" do
+      command lazy { "/usr/sbin/semanage fcontext -d '#{new_resource.file_spec}' && #{restorecon(new_resource.file_spec)}" }
+      only_if "/usr/sbin/semanage fcontext -l | grep -P '#{escaped_file_spec}.*#{new_resource.secontext}'>/dev/null"
```
Comment (BackSlasher) [inline providers/fcontext.rb]: Regex should include spaces (`#{a}\s.*\s#{b}` or something) so it'll be less icky

File: metadata.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
 license          'GPL v2'
 description      'Manages SELinux policy components'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version          '0.5.0'
+version          '0.6.0'
```
Comment (BackSlasher) [inline metadata.rb]: I'd rather do the bumping.
BTW, you didn't update the `CHANGELOG.md`

**Comment by BackSlasher** [comment]
Made some nitpicking comments.
As a side note, could you send me a failing regex so I can play around with it?
Thanks!

**Comment by joerg** [comment]
Hi. Regarding find, this does not work for me (no restults): find /etc -regex "/etc(/.*)?" -regextype posix-basic, while the same regex without brackets works. For this reason and another (see below) I decided to use a recursive call. But I made it easily changeable...
Chef itself will use a recursive restorecon when adding files/templates/directories etc. See here: http://www.rubydoc.info/github/opscode/chef/frames/Chef/Util/Selinux. Look at the code of restore_security_context, they actually __ALWAYS__ use it recursively. I am pretty sure this is a bug, but well.

**Pull request closed without merging**